sources/packages/backend/apps/db-migrations/src/sql/CASInvoices/Create-cas-invoices.sql

```diff
@@ -0,0 +1,43 @@
+CREATE TABLE sims.cas_invoices(
+  id SERIAL PRIMARY KEY,
+  cas_invoice_batch_id INT NOT NULL REFERENCES sims.cas_invoice_batches(id),
+  disbursement_receipt_id INT NOT NULL REFERENCES sims.disbursement_receipts(id),
+  cas_supplier_id INT NOT NULL REFERENCES sims.cas_suppliers(id),
```

Comment:
From SIMS perspective will it be better to have student_id here? So that if there is a change in supplier between invoice generation and invoice being sent, it will be handled? Or will that approach create undesired effects?

Reply:
The student will be retrieved using `disbursement_receipt_id`, no need to be added as a column.
For now the idea would be to keep the `cas_supplier_id` associated during the batch creation to allow the user to review it as part of the snapshot. The `supplier_id` used to generate the batch or even the one used to call the API should be kept for history.
I can double-check if any action is expected in case the supplier number was changed between batch generation and API call. IMO it would just cause a manual intervention only.
If `the supplier_id` does not need to be part of the snapshot approval we can change this column to optional, but being the valid supplier part of the conditions to invoice generation I still feel it must be part of the snapshot.

Reply:
Got it. I agree that `cas_supplier_id` being saved as a part of invoice would be critical to audit what information was sent to CAS.